legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-physical-plan/src/main/java/org/finos/legend/engine/persistence/components/schemaevolution/SchemaEvolution.java

```diff
@@ -174,9 +205,31 @@ private void evolveDataType(Field newField, Field mainDataField, Dataset mainDat
     {
         if (!mainDataField.equals(newField))
         {
-            operations.add(Alter.of(mainDataset, Alter.AlterOperation.CHANGE_DATATYPE, newField, Optional.empty()));
+            if (!mainDataField.type().equals(newField.type()))
+            {
+                operations.add(Alter.of(mainDataset, Alter.AlterOperation.CHANGE_DATATYPE, newField, Optional.empty()));
+                modifiedFields.add(newField);
+            }
+            // todo : tinyint(16) -> int (8) and nullability changes. does above statement handle both ?
+            else if (mainDataField.nullable() != newField.nullable())
```

Comment:
This seems out of place here, we can remove this nullable block from here
and invoke the alterColumnWithNullable directly from the pace where you need to call it